File: projeto/java/LogicalClock.java
public class LogicalClock {

    private int time = 0;

    public synchronized int increment() {
        time++;
        return time;
    }

    public synchronized void update(int received) {
        time = Math.max(time, received);
    }

    public synchronized int getTime() {
        return time;
    }
}

File: projeto/python/LogicalClock.py
import threading

class LogicalClock:
    def __init__(self):
        self.time = 0
        self.lock = threading.Lock()

    def increment(self):
        with self.lock:
            self.time += 1
            return self.time

    def update(self, received):
        with self.lock:
            self.time = max(self.time, received)

    def get_time(self):
        with self.lock:
            return self.time